src/selectors/prevThought.ts

```diff
@@ -0,0 +1,76 @@
+import Path from '../@types/Path'
+import State from '../@types/State'
+import expandThoughts from '../selectors/expandThoughts'
+import { getAllChildrenAsThoughts, isVisible } from '../selectors/getChildren'
+import prevSibling from '../selectors/prevSibling'
+import rootedParentOf from '../selectors/rootedParentOf'
+import appendToPath from '../util/appendToPath'
+import hashPath from '../util/hashPath'
+import head from '../util/head'
+import isRoot from '../util/isRoot'
+import parentOf from '../util/parentOf'
+
+/** Checks if a thought is expanded using the expandThoughts selector. */
+const isExpanded = (state: State, path: Path): boolean => {
+  const expandedPathMap = expandThoughts(state, state.cursor)
+  return !!expandedPathMap[hashPath(path)]
+}
+
+/** Gets the last visible descendant of a thought. */
+const lastVisibleDescendant = (state: State, path: Path): Path => {
+  const id = head(path)
+
+  // Only traverse children if the thought or its parent is expanded
+  if (!isExpanded(state, path)) return path
+
+  // When showHiddenThoughts is true, show all children, otherwise filter hidden ones
+  const children = state.showHiddenThoughts
+    ? getAllChildrenAsThoughts(state, id)
+    : getAllChildrenAsThoughts(state, id).filter(child => isVisible(state, child))
+
+  // If no children, return current path
+  if (children.length === 0) return path
+
+  const lastChild = children[children.length - 1]
+  if (!lastChild) return path
+
+  const childPath = appendToPath(path, lastChild.id)
+  return lastVisibleDescendant(state, childPath)
+}
+
+/** Gets the previous visible sibling, recursively skipping hidden thoughts if needed. */
+const getPreviousVisibleSibling = (state: State, path: Path, currentSibling: Path | null = null): Path | null => {
```

Comment:
It might just be a naming issue, but I'm having a little trouble understanding this function. It says in the name and JSDOC that it gets the previous sibling, but it calls `lastVisibleDescendant` internally, which clearly is not limited to siblings.

Can you explain whether this is indeed supposed to only select siblings (i.e. thoughts with the same parent), or if it has a different purpose? If it has a different purpose, we may need to come up with a better name.

Reply:
- A
  - A1
    - A1a
- B (current thought)

When called on B, it will return A1a, not A.

So, I changed like this : getLastVisibleDescendantOfPreviousSibling